### ISSUE TITLE
New container image blew away database

### ISSUE THREAD

**Issue by alex-phillips** [hay-kot/mealie#99]
Been using mealie now for a couple of weeks and a container pull this weekend blew away all of my recipes. Using mongo FWIW.

**Comment by hay-kot** [comment]
Can you verify that your db_type env variable is set to mongo? You may need to add this to your docker-compose file. 

```yml
    environment:
      db_type: "mongo"
```

**Comment by alex-phillips** [comment]
Got it back up, but now when I go to save a recipe, I get the following error:
```
172.19.0.2:44404 - "GET /api/recipe/Internal%20Server%20Error/ HTTP/1.1" 500
[2021-01-19 19:52:16 -0500] [14] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/uvicorn/protocols/http/httptools_impl.py", line 396, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 201, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 150, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 34, in run_in_threadpool
    return await loop.run_in_executor(None, func, *args)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/app/routes/recipe_routes.py", line 50, in get_recipe
    recipe = Recipe.get_by_slug(recipe_slug)
  File "/app/services/recipe_services.py", line 97, in get_by_slug
    document = db.recipes.get(slug, "slug")
  File "/app/db/db_base.py", line 119, in get
    document = self.document.objects.get(**{str(match_key): match_value})
  File "/usr/local/lib/python3.8/site-packages/mongoengine/queryset/base.py", line 270, in get
    raise queryset._document.DoesNotExist(msg)
db.mongo.recipe_models.DoesNotExist: RecipeDocument matching query does not exist.
```

**Comment by hay-kot** [comment]
I missed adding the new time field to mongo. If you omit them for now you should be able to save them. I'll work on getting a hot-fix out.

**Comment by alex-phillips** [comment]
I did omit those fields, only included ingredients, steps, and an image in that error I posted.

**Comment by hay-kot** [comment]
Found and fixed the bug. New release is being packaged.

**Comment by alex-phillips** [comment]
Thanks! Loving the project so far, btw. Very nice.

**Comment by alex-phillips** [comment]
Still unable to save with latest container. Latest error:

```
172.19.0.47:50192 - "POST /api/recipe/create/ HTTP/1.1" 500
[2021-01-22 18:36:15 -0500] [122] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/uvicorn/protocols/http/httptools_impl.py", line 396, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 201, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 150, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 34, in run_in_threadpool
    return await loop.run_in_executor(None, func, *args)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/app/routes/recipe_routes.py", line 81, in create_from_json
    created_recipe = data.save_to_db()
  File "/app/services/recipe_services.py", line 116, in save_to_db
    recipe_doc = db.recipes.save_new(recipe_dict)
  File "/app/db/db_base.py", line 145, in save_new
    new_document.save()
  File "/usr/local/lib/python3.8/site-packages/mongoengine/document.py", line 378, in save
    self.validate(clean=clean)
  File "/usr/local/lib/python3.8/site-packages/mongoengine/base/document.py", line 433, in validate
    raise ValidationError(message, errors=errors)
mongoengine.errors.ValidationError: ValidationError (RecipeDocument:None) (Field is required and cannot be empty: ['recipeIngredient'])
{"loglevel": "info", "workers": 12, "bind": "0.0.0.0:80", "graceful_timeout": 120, "timeout": 120, "keepalive": 5, "errorlog": "-", "accesslog": "-", "workers_per_core": 1.0, "use_max_workers": null, "host": "0.0.0.0", "port": "80"}
Jobstore default:
    webhooks (trigger: cron[hour='0', minute='0'], next run at: 2021-01-23 00:00:00 EST)
    auto_backup_job (trigger: cron[hour='3'], next run at: 2021-01-23 03:00:00 EST)
172.19.0.47:50194 - "POST /api/recipe/Internal%20Server%20Error/update/image/ HTTP/1.1" 500
[2021-01-22 18:36:15 -0500] [117] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/uvicorn/protocols/http/httptools_impl.py", line 396, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 201, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 150, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 34, in run_in_threadpool
    return await loop.run_in_executor(None, func, *args)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/app/routes/recipe_routes.py", line 92, in update_recipe_image
    Recipe.update_image(recipe_slug, extension)
  File "/app/services/recipe_services.py", line 135, in update_image
    db.recipes.update_image(slug, extension)
  File "/app/db/db_recipes.py", line 63, in update_image
    document = self.document.objects.get(slug=slug)
  File "/usr/local/lib/python3.8/site-packages/mongoengine/queryset/base.py", line 270, in get
    raise queryset._document.DoesNotExist(msg)
db.mongo.recipe_models.DoesNotExist: RecipeDocument matching query does not exist.
172.19.0.47:50196 - "GET /api/recipe/Internal%20Server%20Error/ HTTP/1.1" 500
[2021-01-22 18:36:15 -0500] [117] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/uvicorn/protocols/http/httptools_impl.py", line 396, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 201, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 150, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 34, in run_in_threadpool
    return await loop.run_in_executor(None, func, *args)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/app/routes/recipe_routes.py", line 50, in get_recipe
    recipe = Recipe.get_by_slug(recipe_slug)
  File "/app/services/recipe_services.py", line 97, in get_by_slug
    document = db.recipes.get(slug, "slug")
  File "/app/db/db_base.py", line 119, in get
    document = self.document.objects.get(**{str(match_key): match_value})
  File "/usr/local/lib/python3.8/site-packages/mongoengine/queryset/base.py", line 270, in get
    raise queryset._document.DoesNotExist(msg)
db.mongo.recipe_models.DoesNotExist: RecipeDocument matching query does not exist.
```

**Comment by hay-kot** [comment]
Can you post the JSON of the recipe you're trying to save?

**Comment by alex-phillips** [comment]
@hay-kot This was a brand new recipe, just name, image, and steps. Here is the JSON from the JSON viewer before I hit save:

```json
{
  "name": "Chicken Piccata",
  "description": "",
  "image": "",
  "recipeYield": "",
  "recipeIngredient": [],
  "recipeInstructions": [
    {
      "text": "Butter and oil in pan set near highest setting until it starts to foam (don't let it burn)"
    },
    {
      "text": "Dip chicken slices in bowl of flour to coat with flour"
    },
    {
      "text": "Add more butter/oil to pan if not liquidy (flour with absorb this)"
    },
    {
      "text": "Put chicken in pan. Keep in until lightly browned, the flip."
    },
    {
      "text": "Leave any remnant flour and butter/oil in pan"
    },
    {
      "text": "Add 2-2.5 cups of chicken broth, or 2.5 cups of hot water and 3 tsp of bouillon into pan. Place back on heat."
    },
    {
      "text": "Once liquid soaks up remnants in pan, heat until boiling, then remove from heat."
    },
    {
      "text": "Squeeze lemon(s) into pan."
    },
    {
      "text": "Put pan on low heat. Add chicken. Simmer for 20 minutes."
    },
    {
      "text": "Sprinkle flour into pan to create thick sauce."
    }
  ],
  "slug": "",
  "filePath": "",
  "tags": [],
  "categories": [],
  "notes": [],
  "extras": []
}
```

**Comment by hay-kot** [comment]
Fixed in the [new release v0.2.0](https://github.com/hay-kot/mealie/releases/tag/v0.2.0) 🎉
